src/gui/mainwindow.cpp

```diff
@@ -399,7 +399,23 @@ MainWindow::MainWindow(IGUIApplication *app, WindowState initialState)
         if (!isVisible())
             activate();
 #else
-        toggleVisibility();
+        if (isActiveWindow())
+        {
+            hide();
+        }
+        else
+        {
+            if (m_uiLocked && !unlockUI())  // Ask for UI lock password
+                return;
+
+            // Make sure the window is not minimized
+            setWindowState((windowState() & ~Qt::WindowMinimized) | Qt::WindowActive);
+
+            // Then show it
+            show();
+            raise();
+            activateWindow();
+        }
```

Comment:
I believe you shouldn't do `setWindowState((windowState() & ~Qt::WindowMinimized) | Qt::WindowActive);` manually. The next 3 methods will care about it. Or did your tests show the another result?
And if I'm right, then it can even be rewritten as the following:

```suggestion
        if (isActiveWindow())
            hide();
        else
            activate();
```

Reply:
> Removing only that line or with your suggestion, when qBt is minimized (not to tray) clicking the tray icon doesn't restore the window.

OK.
Seems `setWindowState(windowState() & ~Qt::WindowMinimized)` is really required but setting `Qt::WindowActive` flag does exactly the same as `activateWindow()` so one of them can be omitted.

Unfortunately, this code does not work correctly (at least on Windows, where I managed to test it). It does not hide the main window when clicking on the tray icon. For some reason, `isActiveWindow()` always returns `false`.

Reply:
>but setting Qt::WindowActive flag does exactly the same as activateWindow() so one of them can be omitted.

Ok i see no problems with removing `Qt::WindowActive` but i do with removing `activateWindow()`:
1. qBt out of focus
2. Click tray icon and it brings qBt to the front
3. Subsequent tray icon clicks do nothing
>Unfortunately, this code does not work correctly (at least on Windows, where I managed to test it). It does not hide the main window when clicking on the tray icon. For some reason, isActiveWindow() always returns false.

Damn... i can't test in Windows, i mean i could with CI builds but it would be a chore.

Reply:
> Ok i see no problems with removing `Qt::WindowActive` but i do with removing `activateWindow()`



> Ok i see no problems with removing `Qt::WindowActive` but i do with removing `activateWindow()`

In fact, the order of operations matters. `show()` has to go before `activateWindow()`, otherwise it doesn't work.

Reply:
>For some reason, `isActiveWindow()` always returns `false`.

It looks like main window becomes inactive at the moment when the user interacts with the tray icon.